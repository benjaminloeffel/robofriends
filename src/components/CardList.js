import React from "react";
import PropTypes from "prop-types";
import { Grid } from "@theme-ui/components";
import Card from "../components/Card";

const CardList = ({ users }) => {
  return (
    <>
      <Grid
        gap={3}
        columns={[2, null, 5]}
        sx={{
          bg: "text",
          padding: 16
        }}
      >
        {users.map((user, idx) => (
          <Card
            key={idx}
            id={user.id}
            username={user.username}
            name={user.name}
            email={user.email}
          />
        ))}
      </Grid>
    </>
  );
};

CardList.propTypes = {
  users: PropTypes.array.isRequired
};

export default CardList;
