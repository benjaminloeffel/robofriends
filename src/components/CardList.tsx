import React from "react";
import PropTypes from "prop-types";
import { Grid } from "@theme-ui/components";
import Card from "./Card";
import { User } from "../store/types";

type Props = {
  users: Array<User>;
};

const CardList: React.FC<Props> = ({ users }) => {
  return (
    <>
      <Grid
        gap={3}
        columns={[2, null, 5]}
        sx={{
          bg: "text",
          padding: 16
        }}
      >
        {users.map((user: User, idx: number) => (
          <Card
            key={idx}
            id={user.id}
            username={user.username}
            name={user.name}
            email={user.email}
          />
        ))}
      </Grid>
    </>
  );
};

CardList.propTypes = {
  users: PropTypes.array.isRequired
};

export default CardList;
