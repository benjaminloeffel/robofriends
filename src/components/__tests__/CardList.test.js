import React from "react";
import "../../setupTests";
import { shallow } from "enzyme";
import CardList from "../CardList";

describe("CardList", () => {
  const mockUsers = [
    {
      id: 1,
      name: "John Doe",
      username: "johndoe",
      email: "john@doe.com"
    }
  ];

  it("renders the CardList component with prop", () => {
    const wrapper = shallow(<CardList users={mockUsers} />);
    expect(wrapper).toMatchSnapshot();
  });
});
