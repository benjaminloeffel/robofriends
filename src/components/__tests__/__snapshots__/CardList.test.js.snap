// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList renders the CardList component with prop 1`] = `ShallowWrapper {}`;
